app/src/main/java/com/amaze/filemanager/activities/superclasses/PermissionsActivity.java

```diff
@@ -0,0 +1,113 @@
+package com.amaze.filemanager.activities.superclasses;
```

Comment:
It sounds a bit awkward to me that a superclass is placed below the subclass. However, it makes sense if it is heading for 3.3.0 since the blast radius will be too big if `MainActivity` should be placed below `PermissionsActivity`.

Reply:
What do you mean "below"? Physically below in some visual representation or hierarchically below?

Reply:
Hierarchically below.

Reply:
`MainActivity` extends `PermissionsActivity`, `MainActivity` is hierarchically below `PermissionsActivity`.